### DISCUSSION TITLE
How to disable alpha blending?

### DISCUSSION THREAD

**Discussion by BigheadSMZ** [ImageMagick/ImageMagick#2562] [Help] [answered]
Hello, I have a quick question. Suppose I am doing a straight conversion of PNG to a PNG, that contains pixels with RGB values greater than 0, and the alpha value of those pixels is 0, the RGB channels are all set to 0 after the conversion.  

I was doing a bit of research into this, and I came across this post.
https://www.imagemagick.org/discourse-server/viewtopic.php?p=84108#p84108

There it was said that IM7 would have the ability to disable 'alpha blending' completely to prevent this from happening so the color values would be preserved even when the alpha is 0, but I am not sure what the command would be for this.

**Comment by snibgo** [comment]
What version of IM, on what platform? Please show the exact command you used, and a sample input.

A simple read-write like this...
```
magick in.png out.png
```
... where in.png contains fully transparent pixels will not change pixels to transparent black.

**Comment by BigheadSMZ** [comment]
ImageMagick version used is the most recent: 7.0.10-29. Platform is Windows 10.
As for the command, well ImageMagick is being used in a PowerShell script with a variety of options that can change the overall command, but no matter what I use it always comes out the same. The most basic example that exhibits the issue would probably look something like...

magick in.png -define png:color-type=6 out.png

In my case, it is important to define the png color type to prevent creating a palette image (indexed). To show an example, when hiding the alpha channel in GIMP, this is how the image should look. The transparent pixels hold color values.
https://imgur.com/VM8fq0C

This is how it looks after being processed by ImageMagick.
https://imgur.com/cu7ORuw

Since these images are being used as game textures, it's important to preserve the color data since the game can see it and creates some ugly looking black holes in the textures when that data is absent.

Edit: Can never get image links to work, so just linking instead.

**Comment by snibgo** [comment, marked as answer]
Please link to your in.png. The actual file, not a screenshot of how some program displays the file.

In IM v7.0.10-27, on Windows 8.1, this ...
```
magick c.png -define png:color-type=6 c2.png
```
... does not change transparent pixels to transparent black.

**Comment by BigheadSMZ** [comment]
Thanks for all your help so far. I feel a bit foolish now, it seems that the "-sharpen" command is always being added and I totally overlooked it. I apologize for that as it is what is causing the issue. It looks like the minimum command is actually...

magick in.png -sharpen 0x0 -define png:color-type=6 out.png

I use a very simple formula to calculate the amount of sharpening based on the amount of resolution change from the original image (if applicable -resize is added), which is far from perfect but works fine in most cases. This happens even if the sharpen value is 0x0. I'm starting to think the path of least resistance would be to separate the alpha channel from the image and work with them separately, as increasing the resolution also creates the black spots even without the sharpening filter, albeit less frequently.

Here is the image file as requested:
http://www.mediafire.com/view/uvbo4zkc2hzjw1f/in.png/file#

**Reply by fmw42** [reply]
I am not sure what you are asking, but if you only want to sharpen the RGB channels, then 

`magick in.png -channel rgb -sharpen 0x0 +channel -define png:color-type=6 out.png`

**Comment by BigheadSMZ** [comment]
Sorry for all the trouble, I managed to get it working as intended using:

magick in.png -channel RGBA -separate (+commands) -combine out.png

This allows me to apply filters like sharpening and increase the resolution and the output is what I would expect. For example, the image below has had its resolution increased and the color values have been preserved. Thank you for taking the time to help.

https://i.imgur.com/MaX4JAa.png
